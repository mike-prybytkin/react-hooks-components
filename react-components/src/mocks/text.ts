const mockText = {
  homeLink: 'home',
  aboutLink: 'about',
  cardTitle: 'Fjallraven',
  searchPlaceholder: 'Search...',
  linkBackToHome: 'Back to home page',
  heading404: 'Error 404!',
  headingAboutUs: 'About Us!',
  headingMain: 'Welcome to us!',
  pageNotReady: 'Sorry, page not ready yet...',
  loading: 'Loading ...',
  itemNotFound: 'Sorry, item not found...',
  headingForm: 'Create User!',
  notCreatedUserInForm: 'Sorry, no users have been created yet...',
  labelUserName: 'Name:',
  placeholderUserName: 'Your Name',
  labelUserBirthday: 'Birthday:',
  labelUserSalary: 'Your monthly income:',
  switcherGender: 'Gender:',
  switcherGenderMale: 'Male',
  switcherGenderFemale: 'Female',
  labelUserMailing: 'Receive mailing',
  labelFileUpload: 'Add your photo',
  createUserButton: 'Create user',
  productCardButton: 'Learn more',
  prevButton: 'prev',
  nextButton: 'next',
  userSex: 'sex:',
  userBDay: 'b-day:',
  userSalary: 'salary:',
  subscribed: 'Subscribed',
  unsubscribed: 'Unsubscribed',
  toMailings: 'to mailings!',
  fromMailings: 'from mailings...',
};

export default mockText;
