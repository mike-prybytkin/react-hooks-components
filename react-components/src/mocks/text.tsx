const mockText = {
  homeLink: 'home',
  aboutLink: 'about',
  cardTitle: 'Fjallraven',
  searchPlaceholder: 'Search...',
  linkBackToHome: 'Back to home page',
  heading404: 'Error 404!',
  headingAboutUs: 'About Us!',
  headingMain: 'Welcome to our store!',
  pageNotReady: 'Sorry, page not ready yet...',
  loading: 'Loading ...',
  itemNotFound: 'Sorry, item not found...',
  headingForm: 'Create User!',
  notCreatedUserInForm: 'Sorry, no users have been created yet...',
  labelUserName: 'Name:',
  placeholderUserName: 'Your Name',
  labelUserBirthday: 'Birthday:',
  labelUserSalary: 'Your monthly income:',
  switcherGender: 'Gender:',
  switcherGenderMale: 'Male',
  switcherGenderFemale: 'Female',
  labelUserMailing: 'Receive mailing',
  labelFileUpload: 'Add your photo',
  createUserButton: 'Create user',
};

export default mockText;
