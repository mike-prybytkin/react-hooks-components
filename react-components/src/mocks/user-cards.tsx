import { IUserCard } from 'share/types';
import { userPhoto } from './user-photo';

export const mockUserCards: IUserCard[] = [
  {
    name: 'Max',
    avatarPath: userPhoto.max,
    gender: 'Male',
    birthday: '12-08-1995',
    salary: 'less than 100$',
    mailing: true,
  },
  {
    name: 'Kate',
    avatarPath: userPhoto.kate,
    gender: 'Female',
    birthday: '01-12-2003',
    salary: '200$ - 299$',
    mailing: false,
  },
];
