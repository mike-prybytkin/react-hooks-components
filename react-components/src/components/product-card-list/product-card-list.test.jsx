import React from 'react';
import { render, screen } from '@testing-library/react';
import ProductCardList from './product-card-list';
import { mockCards } from '../../mocks/cards';

const setUp = (props) => render(<ProductCardList cards={props} />);

describe('Card list component', () => {
  it('should render card list without data', () => {
    setUp([]);
    expect(screen.queryByRole('heading')).toBeNull();
  });

  it('should correctly render headings', () => {
    setUp(mockCards);
    expect(screen.queryAllByRole('heading', { level: 3 })).toMatchSnapshot();
  });

  it('create card list snapshot with data', () => {
    const { asFragment } = setUp(mockCards);
    expect(asFragment()).toMatchSnapshot();
  });
});
