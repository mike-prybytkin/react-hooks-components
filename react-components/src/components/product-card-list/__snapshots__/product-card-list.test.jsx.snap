// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card list component create card list snapshot with data 1`] = `
<DocumentFragment>
  <div
    class="cards-wrapper"
  >
    <div
      class="card"
    >
      <img
        alt="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
        class="card__img"
        src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
      />
      <div
        class="card__wrapper-title"
      >
        <h3
          class="card__wrapper-title-text"
        >
          Rick Sanchez
        </h3>
      </div>
      <button
        class="card__button"
      >
        Learn more
      </button>
    </div>
    <div
      class="card"
    >
      <img
        alt="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
        class="card__img"
        src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
      />
      <div
        class="card__wrapper-title"
      >
        <h3
          class="card__wrapper-title-text"
        >
          Morty Smith
        </h3>
      </div>
      <button
        class="card__button"
      >
        Learn more
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card list component should correctly render headings 1`] = `
Array [
  <h3
    class="card__wrapper-title-text"
  >
    Rick Sanchez
  </h3>,
  <h3
    class="card__wrapper-title-text"
  >
    Morty Smith
  </h3>,
]
`;
